### PR TITLE
Remove out/in parameters from function pointers to avoid generating marshalling stubs.

### PR DESCRIPTION
**Pull request by jlaanstra**
Contributes to #1315 and #1318

### CODE REVIEW

**Comment by manodasanW** [comment]
Some of these are already addressed in the staging/aot branch.  But these changes might be worth having in master before the aot branch is merged.  We should do a comparison between the changes here and there and make sure they are done in the same way or any that are missing there are ported.

File: src/cswinrt/code_writers.h [hunk 1]
```diff
@@ -3195,19 +3195,23 @@ db_path.stem().string());
     auto get_invoke_info(writer& w, MethodDef const& method, uint32_t const& abi_methods_start_index = INSPECTABLE_METHOD_COUNT)
     {
         TypeDef const& type = method.Parent();
-        if (!settings.netstandard_compat && distance(type.GenericParam()) == 0)
+        bool signature_has_generic_parameters = abi_signature_has_generic_parameters(w, method_signature{ method });
```
Comment (jlaanstra) [inline src/cswinrt/code_writers.h:3198]: @ManodasanW to check this logic that I ported from staging/AOT.

**Review by Sergio0694 — COMMENTED** [review]
I'm thinking it probably makes sense to wait for #1497 to be merged first, and then change the target branch for this PR to `staging/AOT` and rebase and update it from there 🤔

**Comment by dongle-the-gadget** [comment]
> change the target branch for this PR to staging/AOT

Didn't Mano say that the changes should probably be merged to master before we officially merge the AOT branch?

**Comment by Sergio0694** [comment]
It just feels like a recipe to have a bazillion merge conflicts. I don't really see the point of merging them first. You don't gain much even if you published another patch release from master, just with this, if you don't have all the other improvements from the AOT branch anyway. Just seems like extra (and very annoying) work for no reason. Better ways to spend your time 😅

**Comment by dongle-the-gadget** [comment]
Some of these fixes have been implemented in the AOT branch, so I feel like rebasing this would probably lead to lots of duplicates.

**Comment by Sergio0694** [comment]
You would have to deal with that eventually anyway. My point is, we can at least just do it once, in the AOT branch.

**Comment by jlaanstra** [comment]
The point of this PR would be to bring some of this from staging/AOT to master. Everything should be already in staging/AOT. If something is missing I can see about porting those change to staging/AOT.

**Pull request closed without merging**